src/gmt_api.c

```diff
@@ -3213,2611 +3308,2712 @@ GMT_LOCAL int gmtapi_export_postscript (struct GMTAPI_CTRL *API, int object_ID,
-		if ((I = GMT_Duplicate_Data (GMT->parent, GMT_IS_IMAGE, GMT_DUPLICATE_DATA, I_in)) == NULL) {
-			GMT_Report (GMT->parent, GMT_MSG_ERROR, "Unable to duplicate image! - this is not a good thing and may crash this module\n");
-			(*I_out) = I_in;
-		}
-		else {
-			struct GMT_IMAGE_HIDDEN *IH = gmt_get_I_hidden (I);
-			IH->alloc_mode = GMT_ALLOC_INTERNALLY;
-		}
-		new = true;
-	}
-	else	/* Here we may overwrite the input image and just pass the pointer back */
-		I = I_in;
-
-	/* Here, I is an image we can reallocate the array when expanding the colors */
-
-	h = I->header;
-	data = gmt_M_memory_aligned (GMT, NULL, h->size * 3, unsigned char);	/* The new r,g,b image */
-
-	if (GMT->parent->GMT->current.gdal_read_in.O.mem_layout[0] && strncmp (GMT->parent->GMT->current.gdal_read_in.O.mem_layout, "TRB", 3U) == 0) {	/* Band interleave */
-		strncpy (h->mem_layout, "TRB ", 4);	/* Fill out red, green, and blue bands */
-		for (c = 0; c < 3; c++) off[c] = c * h->size;
-		for (node = 0; node < h->size; node++) {	/* For all pixels, including the pad */
-			index = I->data[node];	/* Pixel index into color table */
-			start_c = index * 4;	/* Start of the r,g,b entry in the color table for this index */
-			for (c = 0; c < 3; c++) data[node+off[c]] = I->colormap[start_c+c];	/* Place r,g,b in separate bands */
+	if ((mode & GMT_CONTAINER_ONLY) == 0) {	/* Also allocate and initialize the x and y vectors unless already present  */
+		if (G_obj->x == NULL) {
+			GH->xy_alloc_mode[GMT_X] = GMT_ALLOC_INTERNALLY;
+			if (GMT->current.io.nc_xarray)	/* Got variable x-array and asked to used this instead */
+				G_obj->x = GMT->current.io.nc_xarray, GMT->current.io.nc_xarray = NULL;
+			else
+				G_obj->x = gmtapi_grid_coord (API, GMT_X, G_obj);	/* Get array of x coordinates */
 		}
-	}
-	else {	/* Pixel interleave */
-		uint64_t k;
-		strncpy (h->mem_layout, "TRP ", 4);	/* Fill out red, green, and blue pixels */
-		for (node = k = 0; node < h->size; node++) {	/* For all pixels, including the pad */
-			index = I->data[node];	/* Pixel index into color table */
-			start_c = index * 4;	/* Start of the r,g,b entry in the color table for this index */
-			for (c = 0; c < 3; c++, k++) data[k] = I->colormap[start_c+c];	/* Place r,g,b in separate bands */
+		if (G_obj->y == NULL) {
+			GH->xy_alloc_mode[GMT_Y] = GMT_ALLOC_INTERNALLY;
+			if (GMT->current.io.nc_yarray)	/* Got variable y-array and asked to used this instead */
+				G_obj->y = GMT->current.io.nc_yarray, GMT->current.io.nc_yarray = NULL;
+			else
+				G_obj->y = gmtapi_grid_coord (API, GMT_Y, G_obj);	/* Get array of y coordinates */
 		}
+<<<<<<< HEAD
 		/* If neither TRB or TRP we call for a changed layout, which may or may not have been implemented */
 		GMT_Change_Layout (GMT->parent, GMT_IS_IMAGE, GMT->parent->GMT->current.gdal_read_in.O.mem_layout, 0, I, NULL, NULL);
+=======
+>>>>>>> 02af19c043... Reading/writing a matrix from/to a grid file (#3848)
```

Comment:
Conflict 1.

Reply:
Looks like **diff** has a bug to me.  This line involves a change to an image variable I, but we are doing grids in this function.  This is completely crazy. It must have found some similar passage and placed it in the wrong spot.  Those lines are found in the function _gmtapi_expand_index_image_, which is not involved in grid reading.  So I guess delete?

Reply:
Obviously, **diff** doesn't do a good job for big changes like this one.